### ISSUE TITLE
_tee has no effect when passing a pipe as _out / _err

### ISSUE THREAD

**Issue by fletom** [amoffat/sh#384]
sh version: 1.12.13

```
In [13]: sh.curl('-I', 'example.com').stdout
Out[13]: b'HTTP/1.1 200 OK\r\nContent-Encoding: gzip\r\nAccept-Ranges: bytes\r\nCache-Control: max-age=604800\r\nContent-Type: text/html\r\nDate: Fri, 12 May 2017 04:09:21 GMT\r\nEtag: "359670651+gzip"\r\nExpires: Fri, 19 May 2017 04:09:21 GMT\r\nLast-Modified: Fri, 09 Aug 2013 23:54:35 GMT\r\nServer: ECS (ewr/15BD)\r\nX-Cache: HIT\r\nContent-Length: 606\r\n\r\n'

In [14]: sh.curl('-I', 'example.com', _out = sys.stdout, _tee = True).stdout
HTTP/1.1 200 OK
Content-Encoding: gzip
Accept-Ranges: bytes
Cache-Control: max-age=604800
Content-Type: text/html
Date: Fri, 12 May 2017 04:09:48 GMT
Etag: "359670651+gzip"
Expires: Fri, 19 May 2017 04:09:48 GMT
Last-Modified: Fri, 09 Aug 2013 23:54:35 GMT
Server: ECS (ewr/15BD)
X-Cache: HIT
Content-Length: 606

Out[14]: b''
```

the same behaviour occurs with `_tee = 'out'` too

**Comment by amoffat** [comment]
This seems to only not be working with when a pipe or tty is passed to `_out`/`_err`, probably due to the fact that if that is the case, we `os.dup2` the pipe/tty's file descriptor *into* the processes's stdout/stderr.  Meaning there is no way for us to intercept the data flowing from the process into the output handler, since it's connected directly.  This seems like a bug.  Thanks for reporting

**Comment by amoffat** [comment]
Fixed on master, version `1.12.14`
